### DISCUSSION TITLE
順序付きリストの揃え、挙動、約物

### DISCUSSION THREAD

**Discussion by yamahige** [w3c/jlreq-d#72] [General]
順序付き（番号付き）リスト（列挙）のスタイルについて、具体例を、JLReqに加えてもう少しあげたほうがよい気がします。本文中の箇条書きでは目立ちませんが、文末の注一覧や文献一覧は数が多くて特徴があらわれやすいと思います。

ざっと見て次がポイントだと思うのですが、いかがでしょうか？

- 順序ラベル（番号）に左揃えがあること
- 順序ラベルの桁数が大きい場合の挙動
- 約物の詰め。(別途、discussion「[約物と行頭／連続](https://github.com/w3c/jlreq-d/discussions/70)」でも取り上げてます)
- 順序ラベルの番号はアラビア数字か？括弧は和文括弧か？

よく見かけるスタイルの例をあげます。学術論文にはオンラインで無料公開されてるものがあるので、そこからあげます。テキストの最後の方の注一覧や文献一覧を見てください。

## 横書き

- [オンライン上でのコンピューティング実習支援システム「CreCoSpeek」の開発と評価 - 情報処理学会](http://id.nii.ac.jp/1001/00216467/)  
横書き。番号は左揃え、左右に角括弧、番号の桁数が大きい場合は右に伸びる。  
「File / Name」の下のリンクをクリックしてPDFファイルをダウンロードして見てください。サーバーが重いので少し待ってください。

-   [情報処理学会のLaTeXスタイルファイルを使って、ワザと順序ラベルの桁数を大きくしたサンプル（PDF）](https://github.com/user-attachments/files/17613460/jsample.pdf)
前記の論文の[スタイル](https://www.ipsj.or.jp/journal/submit/style.html)を試したもの。横書き。既定値で番号は左揃え、左右に角括弧、番号の桁数が大きい場合は右に溢れて箇条内容はそれを回り込む。順序ラベルの桁数が同じでも、箇条内容の先頭が視覚的に揃わない。

- [LaTeXのjlreqドキュメントクラスを使って、ワザと順序ラベルの桁数を大きくしたサンプル（PDF）](https://github.com/user-attachments/files/17613469/list-jlreq.pdf) 
既定値で順序ラベルは右揃え。箇条内容の先頭が視覚的に揃っている。

J-STAGEに掲載されてる論文の多くはPDFですが、一部にHTML版も載っています。「本文（HTML形式）／Full-text HTML」をクリックして見てください。

- [めまいを主訴とした反復音暴露による限局性恐怖症例 - Equilibrium Research](https://doi.org/10.3757/jser.83.88)  
HTML版。横書き。番号は左揃え、番号の右に括弧、番号の桁数が増えると箇条内容に食い込む。HTML + CSSだが、番号を自動生成していない。順序ラベルは最小の幅を持ち、桁数が大きい場合は右に溢れて箇条内容はそれを回り込むように設定されている（CSSで`display: inline-block; min-width: 25px; float: left;`と設定されているのが分かる）。

- [Green Technology for Extraction of Bioactive Compounds from Edible Plants - Journal of Oleo Science](https://doi.org/10.5650/jos.ess24162)  
J-STAGEに掲載されてる英語論文。HTML版。横書き。番号は左揃え(PDF版は右揃え)、番号の右に括弧、番号の桁数が増えると箇条内容に食い込む。HTML + CSSだが、番号は自動生成していない。順序ラベルは最小の幅を持ち、桁数が大きい場合は右に溢れて箇条内容はそれを回り込むように設定されている（CSSで`display: inline-block; min-width: 25px; float: left;`と設定されているのが分かる）。

## 縦書き

- [足利将軍家子弟・室町殿猶子の門寺院入室とその意義 - 史学雑誌](https://doi.org/10.24471/shigaku.130.9_41)  
縦書きの論文。最後に注がまとめてあって、117個、最大3桁ある。番号は縦中横、番号の上下に括弧。箇条内容の先頭が視覚的に揃っている。  
「PDFをダウンロード」をクリックしてPDFファイルをダウンロードして見てください。

- [『権記』所載の一条院出離歌について - 日本文学](https://doi.org/10.20620/nihonbungaku.55.9_35)  
縦書きの論文。番号は縦中横、番号の左右に括弧。箇条内容の先頭が視覚的に揃っている。

- [彰子賛美の真情 - 中古文学](https://doi.org/10.32152/chukobungaku.75.0_21)  
縦書きの論文。番号は縦中横、括弧なし。箇条内容の先頭が視覚的に揃っている。

**Comment by yamahige** [comment]
順序ラベルと箇条本体との関係には -- 明言されたものを見かけませんが -- 整理の仕方が2つあるのではないでしょうか？

- 順序ラベルと箇条本体とは一続きのテキストになっている
- 順序ラベルと箇条本体は分かれている

この違いを約物の観点から見ると
- 順序ラベルと箇条本体が一続きのテキストになっている  
順序ラベル末尾の約物と箇条本体先頭の約物とは**連続約物**である  
リストの組み方は「ぶら下がりインデント」で説明できる(追記 2024-11-04 13:37)
- 順序ラベルと箇条本体は分かれている  
順序ラベル末尾の約物は**行末約物**であり、箇条本体先頭の約物は**行頭約物**である  
リストの組み方を「ぶら下がりインデント」で説明できる…か？(追記 2024-11-04 13:37)

2つの違いは微妙ですが、順序ラベルに用意された（最小）幅が大きくて左揃えだと、後者になってる気がします。

**Comment by kidayasuo** [comment]
番号付きリストの具体例をJLReq-dに加えるのは私も賛成です。縦書きや、ラベルなどに日本語独自の考慮点がありますものね。

**Comment by yamahige** [comment]
かなり途中ですが、次のようなサンプル・ページと図を用意しようとしてます:
[順序付きリストのサンプル](https://yamahige.github.io/study/ordered_list/v0.0/list_sample.html)
「要件」と「HTML+CSSでは、このようになるのではないか？」という話との切り分けに苦慮してますが、別途、レビューを求められてる[How to make list markers stand upright in vertical text](https://w3c.github.io/i18n-drafts/questions/qa-upright-counters-in-vertical.html)に対するコメントとしては、それほど気にしなくてよいかもしれません。

【図】 番号を左揃え、番号はアラビア数字、番号の括弧は欧文括弧、本体の括弧は和文括弧、番号と本体は連続せずそれぞれに行頭・行末がある。
<img width="879" alt="figure-float-number-laten-paren-ja" src="https://github.com/user-attachments/assets/92521aac-0643-4fc0-841c-0f184cf5329a">

【図】 番号を左揃え、番号はアラビア数字、番号の括弧は欧文括弧、本体の括弧は和文括弧、番号と本体は連続する。
<img width="918" alt="figure-inline-block-number-laten-paren-ja" src="https://github.com/user-attachments/assets/8c7e2c03-b8b8-4fde-8d8a-8ec2b91a8918">

**Comment by yamahige** [comment]
縦書きのとき番号の上下に置く括弧に、和文の括弧を使うことについて。

- 約物の詰めは積極的に取り上げた方がよいのでは？
- ラテン文字用の括弧を使うと、括弧の円弧が左にずれることがないか？

# 約物の詰め

最近、Google Chrome、Vivliostyleが約物の詰めをサポートしました。Microsoft Wordもいつのまにかサポートしてます(次の図を参照)。ここは約物の詰めを積極的に取り上げるタイミングではないでしょうか。
ただし、Google ChromeとMicrosoft Wordは、段落の行頭約物は詰められるが、リストの番号の行頭約物は詰められません。これが仕様／未サポート／不具合のどれかは不明です。リストの番号の先頭の約物の詰められるようにしてほしいので、その意味でも和文の括弧を例にあげたいです。
<img width="500" alt="Microsoft Wordで行頭の記号を1/2の幅にする" src="https://github.com/user-attachments/assets/faaa4161-307b-4934-8932-017d2ef86ad5">

# ラテン文字用の括弧

次のような箇条書きを、縦書きで、番号を縦中横にし、番号の上下に括弧を置く場合を検討すると(図を参照):
(10)ああああああああああああああああああああああああ
<img width="500" alt="縦書きの箇条書きの括弧" src="https://github.com/user-attachments/assets/f964cbcc-2bbb-4510-811f-ead35911b803">

次の3つの左右位置が整っていてほしいです:
- (): 上下の括弧
- 10: 縦中横に配置した数字
- ああ…: 本体

ここで、上下の括弧にラテン文字用の括弧を使うと、括弧の円弧が左にずれることがないでしょうか？フォントによるのでしょうが。
JLReq-dのdraft(2024-12-02)で次のように説明してるのと同様なのではないか。
> 括弧には和文用の括弧と、ラテン文字用の括弧があるが、和文の括弧は和字を囲むようにデザインされているのに対し、ラテン文字用の括弧はラテン文字をディセンダを含めて囲むようにデザインされており、下図のように上下位置が異なる。
> [2 文字とフォント > 和文約物と欧文約物](https://github.com/w3c/jlreq-d/blob/gh-pages/drafts/2.%20%E6%96%87%E5%AD%97%E3%81%A8%E3%83%95%E3%82%A9%E3%83%B3%E3%83%88.md#%E5%92%8C%E6%96%87%E7%B4%84%E7%89%A9%E3%81%A8%E6%AC%A7%E6%96%87%E7%B4%84%E7%89%A9)

括弧は和文の括弧にした方が手堅いのではないか？
